kafka-client/src/main/java/dev/responsive/kafka/api/async/AsyncProcessorWrapper.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright 2024 Responsive Computing, Inc.
+ *
+ * This source code is licensed under the Responsive Business Source License Agreement v1.0
+ * available at:
+ *
+ * https://www.responsive.dev/legal/responsive-bsl-10
+ *
+ * This software requires a valid Commercial License Key for production use. Trial and commercial
+ * licenses can be obtained at https://www.responsive.dev
+ */
+
+package dev.responsive.kafka.api.async;
+
+import org.apache.kafka.streams.ProcessorWrapper;
+import org.apache.kafka.streams.processor.api.FixedKeyProcessorSupplier;
+import org.apache.kafka.streams.processor.api.ProcessorSupplier;
+
+public class AsyncProcessorWrapper implements ProcessorWrapper {
+
+
+  @Override
+  public <KIn, VIn, KOut, VOut> ProcessorSupplier<KIn, VIn, KOut, VOut> wrapProcessorSupplier(
+      final String processorName,
+      final ProcessorSupplier<KIn, VIn, KOut, VOut> processorSupplier
+  ) {
+    return AsyncProcessorSupplier.createAsyncProcessorSupplier(processorSupplier);
```

Comment:
I'm wondering if we really want to wrap every processor as async -- WDYT about only wrapping stateful operators by default?

(We can of course still allow users to inject their own custom async processors if they have some heavier processing without state like an RPC)

Reply:
I was thinking we'd leverage `configure()` to allow us to plug into exactly this consideration. Not sure how we want to expose that as an API, but we can figure that out later. For now we can do just stateful ones. Like you said, if it's something stateless they want to make async it's pretty easy to write a processor for it manually.